x-pack/plugins/security/server/analytics/analytics_service.ts

```diff
@@ -42,6 +72,200 @@ export interface AuthenticationTypeAnalyticsEvent {
   httpAuthenticationScheme?: string;
 }
 
+/**
+ * Properties that all Reporting API schemas share.
+ */
+interface CommonReportFields {
+  type: string;
+  age?: number;
+  body: {};
+}
+
+/**
+ * Helper type that transforms any Reporting API schema into its corresponding EBT schema:
+ *
+ * - Removes `type` property since events are identified by their `eventType` in EBT.
+ * - Replaces `age` property with `created` timestamp so that we capture a fully qualified date.
+ * - Spreads `body` property to keep the resulting EBT schema flat.
+ */
+type FlattenReport<T extends CommonReportFields> = { created: string } & Omit<
+  T,
+  keyof CommonReportFields
+> &
+  T['body'];
+
+/**
+ * Describes the shape of the CSP violation event type.
+ */
+const cspViolation: EventTypeOpts<CSPViolationEvent> = {
+  eventType: CSP_VIOLATION_EVENT_TYPE,
+  schema: {
+    created: {
+      type: 'keyword',
+      _meta: {
+        description: 'Type of the Kibana authentication provider.',
```

Comment:
It looks like the description strings are a copy/paste typo (all the same). Same for `permissionsPolicyViolation` below.

Reply:
Updated